src/bin/polymer-bundler.ts

```diff
@@ -224,59 +228,94 @@ if (options.redirect) {
+        r.prefix,
+        resolvedPath);
+  });
+
+  // Wrap the getPackageRelativeUrl function for each redirection.  Because we
+  // are wrapping the function, we iterate in reverse order to ensure the
+  // redirect transformations are processed in the same order as they're given.
+  redirections.reverse().forEach((r: redirection) => {
+    const oldGetPackageRelativeUrl = getPackageRelativeUrl;
+    const newGetPackageRelativeUrl = function(resolvedUrl: ResolvedUrl):
+        PackageRelativeUrl {
+          const redirectionPathUrl =
+              ensureTrailingSlash(Uri.file(path.resolve(r.path)).toString());
+          if (resolvedUrl.startsWith(redirectionPathUrl)) {
+            return r.prefix + resolvedUrl.slice(redirectionPathUrl.length) as
+                PackageRelativeUrl;
+          }
+          return oldGetPackageRelativeUrl(resolvedUrl);
+        };
+    getPackageRelativeUrl = newGetPackageRelativeUrl;
+  });
+
   const loaders: UrlLoader[] = redirections.map(
-      (r: redirection) =>
-          new PrefixedUrlLoader(r.prefix, new FSUrlLoader(r.path)));
-  if (loaders.length > 0) {
+      (r: redirection) => new FSUrlLoader(path.resolve(r.path)));
+  if (redirections.length > 0) {
     options.analyzer = new Analyzer({
-      urlResolver:
-          new MultiUrlResolver([...resolvers, new PackageUrlResolver()]),
-      urlLoader:
-          new MultiUrlLoader([...loaders, new FSUrlLoader(projectRoot)])
+      urlResolver: new MultiUrlResolver([...resolvers, packageUrlResolver]),
+      urlLoader: new MultiUrlLoader([...loaders, fsUrlLoader]),
     });
   }
 }
 
 if (!options.analyzer) {
-  options.analyzer = new Analyzer({urlLoader: new FSUrlLoader(projectRoot)});
+  options.analyzer = new Analyzer({
+    urlResolver: packageUrlResolver,
+    urlLoader: fsUrlLoader,
+  });
 }
 
 if (options.shell) {
-  options.strategy = generateShellMergeStrategy(options.shell, 2);
+  options.strategy =
+      generateShellMergeStrategy(options.analyzer.resolveUrl(options.shell), 2);
 }
 
 interface JsonManifest {
-  [entrypoint: string]: UrlString[];
+  [entrypoint: string]: PackageRelativeUrl[];
 }
+(async () => {
 
-function bundleManifestToJson(manifest: BundleManifest): JsonManifest {
-  const json: JsonManifest = {};
-  const missingImports: Set<string> = new Set();
-  for (const [url, bundle] of manifest.bundles) {
-    json[url] = [...new Set([
-      // `files` and `inlinedHtmlImports` will be partially duplicative, but use
-      // of both ensures the basis document for a file is included since there
-      // is no other specific property that currently expresses it.
-      ...bundle.files,
-      ...bundle.inlinedHtmlImports,
-      ...bundle.inlinedScripts,
-      ...bundle.inlinedStyles
-    ])];
+  // Produces a bundle manifest object where all the urls represented are
+  // package relative.
+  function bundleManifestToJson(manifest: BundleManifest): JsonManifest {
```

Comment:
It can except we need a way to get package relative URLs, so once I can get those from the Analyzer's UrlResolver, the resolver can be passed in as an argument and then I can move it to the BundleManifest.